monitor/services/src/main/java/org/datacleaner/monitor/server/controllers/DatastoreQueryController.java

```diff
@@ -49,35 +55,167 @@
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.util.UriComponentsBuilder;
 
 @Controller
 public class DatastoreQueryController {
 
-    private static final Logger logger = LoggerFactory.getLogger(DatastoreQueryController.class);
+    private static final Logger logger = LoggerFactory
+            .getLogger(DatastoreQueryController.class);
 
     @Autowired
     TenantContextFactory _tenantContextFactory;
+    
+    @Autowired
+    private ApplicationEventPublisher eventPublisher;
 
     @RolesAllowed(SecurityRoles.TASK_QUERY)
     @RequestMapping(value = "/{tenant}/datastores/{datastore}.query", method = RequestMethod.POST, produces = {
             "text/xml", "application/xml", "application/xhtml+xml", "text/html" })
-    public void queryDatastorePost(HttpServletResponse response, @PathVariable("tenant") final String tenant,
-            @PathVariable("datastore") String datastoreName, @RequestBody String query) throws IOException {
-        queryDatastore(tenant, datastoreName, query, response);
+    public void queryDatastorePost(HttpServletResponse response,
```

Comment:
Looks like your code formatter has a very narrow line width. We normally use 120 chars line width which gives room for a bit more code per line and a few less linebreaks in expressions.